src/vmainwindow.cpp

```diff
@@ -1974,9 +1974,11 @@ void VMainWindow::updateActionsStateFromTab(const VEditTab *p_tab)
     setActionsEnabled(m_editToolBar, file && editMode);
 
     // Handle heading sequence act independently.
-    m_headingSequenceAct->setEnabled(isHeadingSequenceApplicable());
+    m_headingSequenceAct->setEnabled(editMode && file->isModifiable()
```

Comment:
I make this button enabled in read mode to let user have a chance to stop adding sequence to headings when entering read mode. That is, before entering read mode, de-select this button, then VNote won't change your headings when you enter read mode.

Hence I prefer to leave it untouched.